clmm/theory/parent_class.py

```diff
@@ -587,7 +591,132 @@ def eval_surface_density_2h(self, r_proj, z_cl, halobias=1., lsteps=500):
+                  It can be individual redshifts for each source galaxy when `z_src` is an
+                  arrayor all sources are at the same redshift when `z_src` is a float.
+
+                * 'distribution' : A redshift distribution function is provided by `z_src`.
+                  `z_src` must be a one dimentional function.
+
+                * 'beta' : The averaged lensing efficiency is provided by `z_src`.
+                  `z_src` must be a tuple containing
+                  ( :math:`\langle \beta_s \rangle, \langle \beta_s^2 \rangle`),
+                  the lensing efficiency and square of the lensing efficiency averaged over
+                  the galaxy redshift distribution repectively.
+
+                    .. math::
+                        \langle \beta_s \rangle = \left\langle \frac{D_{LS}}{D_S}\frac{D_\infty}
+                        {D_{L,\infty}}\right\rangle
+
+                    .. math::
+                        \langle \beta_s^2 \rangle = \left\langle \left(\frac{D_{LS}}
+                        {D_S}\frac{D_\infty}{D_{L,\infty}}\right)^2 \right\rangle
+
+        beta_kwargs: None, dict
+            Extra arguments for the `compute_beta_s_mean, compute_beta_s_square_mean` functions.
+            Only used if `z_src_info='distribution'`. Possible keys are:
+
+                * 'zmin' (None, float) : Minimum redshift to be set as the source of the galaxy
+                  when performing the sum. (default=None)
+                * 'zmax' (float) : Maximum redshift to be set as the source of the galaxy
+                  when performing the sum. (default=10.0)
+                * 'delta_z_cut' (float) : Redshift interval to be summed with `z_cl` to return
+                  `zmin`. This feature is not used if `zmin` is provided. (default=0.1)
+
+        Returns
+        -------
+        array_like, float
+            The averaged lensing efficiency.
+        """
+        if z_src_info=='beta':
+            # z_src (tuple) is (beta_s_mean, beta_s_square_mean)
+            beta_s_mean = z_src[0]
+        elif z_src_info=='distribution':
+            # z_src (function) if PDZ
+            beta_kwargs = {} if beta_kwargs is None else beta_kwargs
+            beta_s_mean = compute_beta_s_mean(
+                z_cl, self.z_inf, self.cosmo, z_distrib_func=z_src, **beta_kwargs)
+        return beta_s_mean
+
+    def _get_beta_s_square_mean(self, z_cl, z_src, z_src_info='discrete',
+                                beta_kwargs=None):
+        r"""Get mean value of the square geometric lensing efficicency ratio from typical class
+        function.
+
+        Parameters
+        ----------
+        z_cl : float
+            Galaxy cluster redshift
+        z_src : array_like, float, function
+            Information on the background source galaxy redshift(s). Value required depends on
+            `z_src_info` (see below).
+        z_src_info : str, optional
```

Comment:
Same docstring comments as before for the whole new redshift-related docstrings.